src/schema/v2/infiniteDiscovery/deleteDiscoveryArtworkReferencesMutation.ts

```diff
@@ -0,0 +1,164 @@
+  fields: () => ({
+    mutationError: {
+      type: GravityMutationErrorType,
+      resolve: (err) => err,
+    },
+  }),
+})
+
+const ResponseOrErrorType = new GraphQLUnionType({
+  name: "DeleteDiscoveryUserReferencesResponseOrError",
+  types: [SuccessType, FailureType],
+})
+
+export const DeleteDiscoveryUserReferencesMutation = mutationWithClientMutationId<
+  { userId: string },
+  any,
+  ResolverContext
+>({
+  name: "DeleteDiscoveryUserReferencesMutation",
+  description: "Deletes a user artwork references in weaviate",
+  inputFields: {
+    userId: {
+      description: "The user's ID",
+      type: GraphQLNonNull(GraphQLString),
+    },
+  },
+  outputFields: {
+    deleteDiscoveryUserReferencesResponseOrError: {
+      type: ResponseOrErrorType,
+      description: "On success: return boolean. On failure: MutationError.",
+      resolve: (result) => {
+        return result
+      },
+    },
+  },
+  mutateAndGetPayload: async (
+    { userId },
+    { weaviateGraphqlLoader, weaviateDeleteObjectLoader }
+  ) => {
+    if (!weaviateGraphqlLoader || !weaviateDeleteObjectLoader) {
+      new Error("Weaviate loader not available")
+    }
+
+    const query = gql`
+      {
+        Get {
+          InfiniteDiscoveryUsers(
+            where: { path: ["internalID"], operator: Equal, valueString: "${userId}" }
+          ) {
+            _additional {
+              id
+            }
+            likedArtworks {
+              ... on InfiniteDiscoveryArtworks {
+                _additional {
+                  id
+                }
+              }
+            }
+            dislikedArtworks {
+              ... on InfiniteDiscoveryArtworks {
+                _additional {
+                  id
+                }
+              }
+            }
+          }
+        }
+      }
+    `
+    try {
+      const body = await weaviateGraphqlLoader({ query })()
+      const users = body.data.Get.InfiniteDiscoveryUsers
+
+      if (!users.length) {
+        return { success: true }
+      }
+
+      const {
+        _additional: { id: uuid },
+        likedArtworks,
+        dislikedArtworks,
+      } = users[0]
+
+      const references = [
+        ...getArtworkReferences({
+          data: likedArtworks,
+          reference: "likedArtworks",
+          uuid,
+        }),
+        ...getArtworkReferences({
+          data: dislikedArtworks,
+          reference: "dislikedArtworks",
+          uuid,
+        }),
+      ]
+
+      await Promise.all(
+        references.map((ref) =>
+          weaviateDeleteObjectLoader(ref.url, { beacon: ref.beacon })
+        )
+      )
+
+      return { success: true }
+    } catch (error) {
+      const formattedErr = formatGravityError(error)
+      if (formattedErr) {
+        return {
+          ...formattedErr,
+          _type: "GravityMutationError",
+        }
+      } else {
+        throw new Error(error)
+      }
+    }
+  },
+})
+
+const getArtworkReferences = ({ data, reference, uuid }) =>
```

Comment:
Not totally clear why we're doing this on the delete? Wouldn't we want to guard against double entries at the `create` stage of the cross reference?

Also, if we do need to do this here, I wonder if we can't simplify this method a bit? The nested iterators makes it a bit tough to read.

Reply:
> Not totally clear why we're doing this on the delete? Wouldn't we want to guard against double entries at the create stage of the cross-reference?

In this case, the goal during deletion is to ensure that all associated references (liked and disliked) are removed properly. The function:

1. Finds unique entries.
2. Prevents duplicate calls (in case duplicates were added via Weaviate GraphQL queries for example).

Added here to remind us that Weaviate isn't handling duplicate prevention automatically, so we need to manage it either here (I agree, it's not ideal) or at the create stage, or create a another way like a proxy API